DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/component_interfaces_comparison/csr-pipes/src/vector_add.cpp

```diff
@@ -0,0 +1,121 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class SimpleVAddPipes;
+
+// Forward declare pipe names to reduce name mangling (maybe get rid of; pipes
```

Comment:
Turn this into PR comment instead of code comment - do not want to accidentally check that in.
```suggestion
// Forward declare pipe names to reduce name mangling
```